src/libraries/System.Private.CoreLib/src/System/Collections/Generic/NonRandomizedStringEqualityComparer.cs

```diff
@@ -12,23 +13,71 @@ namespace System.Collections.Generic
     // randomized string hashing.
     [Serializable] // Required for compatibility with .NET Core 2.0 as we exposed the NonRandomizedStringEqualityComparer inside the serialization blob
     // Needs to be public to support binary serialization compatibility
-    public sealed class NonRandomizedStringEqualityComparer : EqualityComparer<string?>, ISerializable
+    public sealed class NonRandomizedStringEqualityComparer : INonRandomizedEqualityComparer<string?>, ISerializable
     {
-        internal static new IEqualityComparer<string?> Default { get; } = new NonRandomizedStringEqualityComparer();
+        internal static readonly IEqualityComparer<string?> AroundDefaultComparer = new NonRandomizedStringEqualityComparer(wrapsOrdinalComparer: false);
```

Comment:
`Around` in the name looks odd to me. Should the names match StringComparer - `Ordinal` and `OrdinalIgnoreCase` ?

Reply:
Ok, I see what this means now.

I would not describe the relationship between NonRandomized and regular comparer as one wrapping the other. There is no pointer from the NonRandomized to the regular one.

Reply:
FWIW, Around confused me as well.